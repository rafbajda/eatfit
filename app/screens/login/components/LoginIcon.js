import React from 'react';
import { Thumbnail } from 'native-base';
import { CenterRow } from '../../../shared/styles/common';
import styles from '../styles/loginIconStyles';

const logoImage = require('../../../assets/images/eatfit_full.png');

const LoginIcon = props => {
    const { hidden, iconStyle } = { ...props, ...styles };
    if (hidden) {
        return null;
    }
    return (
        <CenterRow>
            <Thumbnail square source={logoImage} style={iconStyle} />
        </CenterRow>
    );
};

export default LoginIcon;
