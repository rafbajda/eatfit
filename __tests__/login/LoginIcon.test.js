import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import LoginIcon from "../../app/screens/login/components/LoginIcon";

describe('LoginIcon snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<LoginIcon />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
