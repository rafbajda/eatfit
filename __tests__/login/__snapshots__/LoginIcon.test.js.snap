// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginIcon snapshot renders component 1`] = `
<View
  className="sc-bxivhb dulVTN"
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Image
    source={1}
    square={true}
    style={
      Array [
        Object {
          "borderRadius": 0,
          "height": 56,
          "width": 56,
        },
        Object {
          "height": 180,
          "marginBottom": 25,
          "width": 180,
        },
      ]
    }
  />
</View>
`;
